packages/editor/src/plugins/image-gallery/editor.tsx

```diff
@@ -34,31 +39,44 @@ export function ImageGalleryEditor(props: ImageGalleryProps) {
   )
 
   function handleAddImagesButtonClick() {
-    if (state.images.length === 0) {
-      state.images.insert(0, { plugin: EditorPluginType.Image })
-    }
+    state.images.insert(state.images.length, { plugin: EditorPluginType.Image })
+    setCurrentImageIndex(state.images.length)
     setIsModalOpen(true)
   }
 
   function handleMultipleImageUpload(files: File[]) {
-    for (const file of files) {
-      const newImagePluginState = imagePlugin.onFiles?.([file])
-      const newImagePlugin = {
-        plugin: EditorPluginType.Image,
-        state: newImagePluginState?.state as unknown,
+    if (state.images.length + files.length > MAX_IMAGES) {
+      showToastNotice(imageGalleryStrings.tooManyImagesMessage, 'warning')
+    } else {
+      for (const file of files) {
+        const newImagePluginState = imagePlugin.onFiles?.([file])
+        const newImagePlugin = {
+          plugin: EditorPluginType.Image,
+          state: newImagePluginState?.state as unknown,
+        }
+        state.images.insert(currentImageIndex, newImagePlugin)
       }
-      state.images.insert(currentImageIndex, newImagePlugin)
     }
+
     setHasImages(true)
     setIsModalOpen(false)
     dispatch(focus(id))
   }
 
+  function pruneEmptyImagePlugins() {
```

Comment:
when/how do empty images happen?

Reply:
maybe we should not init the plugin with one empty image plugin in the beginning, is that the problem?

Reply:
In order to show the empty Image plugin in the modal, we need to have an empty Image plugin in the state.

When the user closes the modal without uploading anything, we get rid of that plugin using this function. I wouldn't say there's a problem at all, it's just how it works.

Do you see a better flow for this?

Reply:
Nice, thanks for the explanation. Maybe we can call it removeDefaultImagePlugin or something then?

Reply:
Good point, I also changed the function itself to match the name, we don't really need to iterate. Thanks!